### PR TITLE
Merge Shared Preferences

### LINKED ISSUE #694: Merge Shared Preferences
Currently we have two shared preferences.
- Default
- Internal

I vouch to merge those two, just to have less confusion on which one to use. That way we can remove some functions from the Utils Class.

### PR DESCRIPTION
**Pull request by kordianbruck**
Whyever we had two shared prefs - this PR should merge them both into one unified SP.

Fixes #694

### CODE REVIEW

File: app/src/main/java/de/tum/in/tumcampusapp/utils/Utils.java [hunk 1]
```diff
@@ -50,6 +51,43 @@ private Utils() {
...
+            }
+            if (object instanceof Integer) {
+                prefsEditor.putInt(entry.getKey(), (Integer) object);
+            }
+            if (object instanceof Float) {
+                prefsEditor.putFloat(entry.getKey(), (Float) object);
+            }
+            if (object instanceof Long) {
+                prefsEditor.putLong(entry.getKey(), (Long) object);
+            }
```
Comment (pfent) [inline app/src/main/java/de/tum/in/tumcampusapp/utils/Utils.java]: Ain't nobody got Double values :see_no_evil:

File: app/src/main/java/de/tum/in/tumcampusapp/utils/Utils.java [hunk 2]
```diff
@@ -96,15 +134,78 @@ public static String cutText(String text, String startString, String endString)
...
+     * @return setting value, defaultVal if undefined
      */
     public static String getSetting(Context c, String key, String defaultVal) {
         SharedPreferences sp = PreferenceManager.getDefaultSharedPreferences(c);
         return sp.getString(key, defaultVal);
     }
 
     /**
-     * Get a value from the default shared preferences
+     * Get a value from the default shared preferences.
```
Comment (pfent) [inline app/src/main/java/de/tum/in/tumcampusapp/utils/Utils.java]: All those shared preference accesses probably can be replaced with [Android KTX](https://github.com/android/android-ktx) in the near future...

File: app/src/main/java/de/tum/in/tumcampusapp/utils/Utils.java (hunk 1 above)
Comment (kordianbruck) [reply app/src/main/java/de/tum/in/tumcampusapp/utils/Utils.java]: Nope :dagger:

File: app/src/main/java/de/tum/in/tumcampusapp/utils/Utils.java (hunk 2 above)
Comment (kordianbruck) [reply app/src/main/java/de/tum/in/tumcampusapp/utils/Utils.java]: You still have to call `.edit` with that, but yea - it still will be some time till that reaches general availability.

**Comment by kordianbruck** [comment]
@pfent ready to merge then?

**Comment by pfent** [comment]
sure the migration path looks good to me, we only need to think about when we should remove that for future versions

**Pull request merged**